Adjustments/dependencies (Dependencies, BaseController, AuthService)

KianMiridoozini opened the pull request:
adjustments/dependencies

Basecontroller Added

The basecontroller serves as a base for common dependencies used in controllers.

Authservice added.
Some logic related to authentication has moved to authservice.

Those logic were previously implemented in authcontroller. Which meant that controllers had to instantiate new userservice and new authcontroller to access them. OR to rely on static functions.

In order to maintaining project standards, the logic is now called from authservice.

KianMiridoozini committed:
adjustments/dependencies:

All dependencies are now included and handled through dependencies.php

Clean up of redundant require statements.

Only service functions called in controllers. (except static functions from AuthController)

KianMiridoozini committed:
Merge branch 'develop' into adjustments/dependencies

KianMiridoozini committed:
adjustments/dependencies

Basecontroller Added

The basecontroller serves as a base for common dependencies used in controllers.

Authservice added.
Some logic related to authentication has moved to authservice.

Those logic were previously implemented in authcontroller. Which meant that controllers had to instantiate new userservice and new authcontroller to access them. OR to rely on static functions.

In order to mainting project standards, the logic is now called from authservice.

jayb2302 commented on app/services/UserService.php:
```diff
@@ -1,13 +1,16 @@
 class UserService
 {
     private $userRepository;
+    private $badgeRepository;
```
Why do we need badgeRepository both here and in badgeservice ?

KianMiridoozini replied:
because we moved some functions from badgeService to userService, since we don't want to have badgService instantiated in CocktailController

the functions moved to userService related to badges, require access to badgeRepository.

jayb2302 replied:
If we have private badgeService instead of badgeRepository, we create more cohesive userService that combines the user data and their badges.

KianMiridoozini replied:
in that case a function like badge notification will have the following flow
badgeRepository->badgeService->userService->dependencies->basecontroller->cocktailController

jayb2302 replied:
What do you mean?  BadgeRepository is a data access layer that intereacts with the database, but badge service is a business logic layer that uses badgeRepository already, I think that userService should not directly interact with the badgeRepository because that is just mixing responsabilities

jayb2302 commented on app/config/dependencies.php:
```diff
@@ -75,41 +100,85 @@
 );
+$tagService = new TagService(
+    $tagRepository,
+    $ingredientRepository
```
So here aswell, using the ingredientService and vice vesa with tagService, but thats also on me.

KianMiridoozini committed:
adjustments/dependencies (fixed)

Removed static functions

implemented a more dynamic approach for checking if user is admin in controllers

removed some redundant data

fixed comments

and some more, can't remember it's 6am

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 committed:
fix: correct path for BaseController inclusion

jayb2302 reviewed (approved):
Awesome work!
